oaknational/Oak-Web-Application
# Feat/lesq 1185/unit downloads [LESQ-1185]

## Description
This combines LESQ-1185 and LESQ-1196

- Add a UnitDownloadButton component with states for feature flag variants, signed in status and download status 
- Create helper fns for unit downloads similar to lesson download fns

The experiement for this can be seen [here](https://eu.posthog.com/project/124/experiments/28199) on posthog


## How to test

1. Go to https://deploy-preview-3046--oak-web-application.netlify.thenational.academy/teachers/programmes/geography-primary-ks1/units/local-area-where-do-we-go-to-school/lessons
2. Which button you see will depend on which cohort of the experiment you have been placed in, you need to add yourself to [this feature flag](https://eu.posthog.com/project/124/feature_flags/43245) in either group (without adding yourself you will be placed in the control group)
3. In the control group, or if the unit download does not exist, the button should not show
4. When signed out: clicking the unit download button should navigate you to the sign in journey, then redirect to this page
option a:
<img width="600" alt="Screenshot 2024-12-02 at 12 14 15" src="https://github.com/user-attachments/assets/291c3fcc-25a2-4cb4-8c16-6bb3491722cf">  

option b:
<img width="600" alt="Screenshot 2024-12-02 at 14 14 54" src="https://github.com/user-attachments/assets/a104e7d2-ceab-40bc-a91c-24cc02dcc1e9">  

5. When signed in: clicking the button should trigger the download to start, the button will update with a loading spinner and a message will show
<img width="600" alt="Screenshot 2024-12-02 at 12 14 32" src="https://github.com/user-attachments/assets/fe912001-803d-43d4-bfa3-99d3f7f66c7a">  
<img width="600" alt="Screenshot 2024-12-02 at 12 14 41" src="https://github.com/user-attachments/assets/17d87818-0855-42f5-b168-356975271ed0">  

7. An error state should show when there is an error
<img width="600" alt="Screenshot 2024-12-02 at 12 15 59" src="https://github.com/user-attachments/assets/bf3f8a6e-062f-4b79-b499-fbc0cd01869f">
